pkg/kubelet/dockershim/network/cni/cni.go

```diff
@@ -117,18 +118,31 @@ func ProbeNetworkPlugins(confDir string, binDirs []string) []network.NetworkPlug
 	return []network.NetworkPlugin{plugin}
 }
 
-func getDefaultCNINetwork(confDir string, binDirs []string) (*cniNetwork, error) {
-	files, err := libcni.ConfFiles(confDir, []string{".conf", ".conflist", ".json"})
-	switch {
-	case err != nil:
-		return nil, err
-	case len(files) == 0:
-		return nil, fmt.Errorf("No networks found in %s", confDir)
+func getDefaultCNINetwork(confDirs []string, binDirs []string) (*cniNetwork, error) {
+	var files []string
+
+	for _, confDir := range confDirs {
+		f, err := libcni.ConfFiles(confDir, []string{".conf", ".conflist", ".json"})
+		if err != nil {
+			return nil, err
+		}
+
+		files = append(files, f...)
+	}
+
+	if len(files) == 0 {
+		return nil, fmt.Errorf("No networks found in %s", confDirs)
 	}
 
-	sort.Strings(files)
+	// Order by conf filename, not by parent directory name.
```

Comment:
if the same filename exists in two directories, it looks like it is undefined which one this sort puts first